src/model/CoilCoolingDXCurveFitPerformance.cpp

```diff
@@ -0,0 +1,473 @@
+  setUnitInternalStaticAirPressure(773.3);
+  setCapacityControlMethod("Discrete");
+  setEvaporativeCondenserBasinHeaterCapacity(0.0);
+  setEvaporativeCondenserBasinHeaterSetpointTemperature(2.0);
+  {
+    auto schedule = model.alwaysOnDiscreteSchedule();
+    setEvaporativeCondenserBasinHeaterOperatingSchedule(schedule);
+  }
+  setCompressorFuelType("Electricity");
+
+  setBaseOperatingMode(baseOperatingMode);
+}
+
+CoilCoolingDXCurveFitPerformance::CoilCoolingDXCurveFitPerformance(const Model& model,
+                                                                   const CoilCoolingDXCurveFitOperatingMode& baseOperatingMode,
+                                                                   const CoilCoolingDXCurveFitOperatingMode& alternativeOperatingMode1)
+  : ResourceObject(CoilCoolingDXCurveFitPerformance::iddObjectType(),model)
+{
+  OS_ASSERT(getImpl<detail::CoilCoolingDXCurveFitPerformance_Impl>());
+
+  setCrankcaseHeaterCapacity(0.0);
+  setMinimumOutdoorDryBulbTemperatureforCompressorOperation(-25.0);
+  setMaximumOutdoorDryBulbTemperatureforCrankcaseHeaterOperation(10.0);
+  setUnitInternalStaticAirPressure(773.3);
+  setCapacityControlMethod("Discrete");
+  setEvaporativeCondenserBasinHeaterCapacity(0.0);
+  setEvaporativeCondenserBasinHeaterSetpointTemperature(2.0);
+  {
+    auto schedule = model.alwaysOnDiscreteSchedule();
+    setEvaporativeCondenserBasinHeaterOperatingSchedule(schedule);
+  }
+  setCompressorFuelType("Electricity");
+
+  setBaseOperatingMode(baseOperatingMode);
+  setAlternativeOperatingMode1(alternativeOperatingMode1);
+}
+
+CoilCoolingDXCurveFitPerformance::CoilCoolingDXCurveFitPerformance(const Model& model,
+                                                                   const CoilCoolingDXCurveFitOperatingMode& baseOperatingMode,
+                                                                   const CoilCoolingDXCurveFitOperatingMode& alternativeOperatingMode1,
+                                                                   const CoilCoolingDXCurveFitOperatingMode& alternativeOperatingMode2)
+  : ResourceObject(CoilCoolingDXCurveFitPerformance::iddObjectType(),model)
+{
+  OS_ASSERT(getImpl<detail::CoilCoolingDXCurveFitPerformance_Impl>());
+
+  setCrankcaseHeaterCapacity(0.0);
+  setMinimumOutdoorDryBulbTemperatureforCompressorOperation(-25.0);
+  setMaximumOutdoorDryBulbTemperatureforCrankcaseHeaterOperation(10.0);
+  setUnitInternalStaticAirPressure(773.3);
+  setCapacityControlMethod("Discrete");
+  setEvaporativeCondenserBasinHeaterCapacity(0.0);
+  setEvaporativeCondenserBasinHeaterSetpointTemperature(2.0);
+  {
+    auto schedule = model.alwaysOnDiscreteSchedule();
+    setEvaporativeCondenserBasinHeaterOperatingSchedule(schedule);
+  }
+  setCompressorFuelType("Electricity");
+
+  setBaseOperatingMode(baseOperatingMode);
+  setAlternativeOperatingMode1(alternativeOperatingMode1);
+  setAlternativeOperatingMode2(alternativeOperatingMode2);
+}
```

Comment:
Not sure I see actual value in providing these two additional Ctors. It's a lot of code duplication while one can as easily do:

```c++
c = CoilCoolingDXCurveFitPerformance(model, baseMode);
c.setAlternativeOperatingMode1(alternativeOperatingMode1);
c.setAlternativeOperatingMode2(alternativeOperatingMode2);
```

Reply:
Resolved, curves are now fully optional and there is a ctor that takes model in.